packages/code/src/Code.tsx

```diff
@@ -80,6 +92,121 @@ function getWrapperVariantStyle(variant: Variant): string {
   `;
 }
 
+function getSidebarVariantStyle(variant: Variant): string {
+  const colors = variantColors[variant];
+
+  switch (variant) {
+    case 'light':
+      return css`
+        border-color: ${colors[1]};
+        background-color: white;
+      `;
+    case 'dark':
+      return css`
+        border-color: ${colors[1]};
+        background-color: ${colors[1]};
+      `;
+  }
+}
+
+function getCopyButtonStyle(variant: Variant, copied: boolean): string {
+  if (copied) {
+    return css`
+      align-self: center;
+      color: ${uiColors.white};
+      background-color: ${uiColors.green.base};
+    `;
+  }
+
+  if (variant === Variant.Dark) {
+    return css`
+      align-self: center;
+      color: ${uiColors.gray.base};
+      background-color: ${uiColors.gray.dark3};
+    `;
+  }
+
+  return css`
+    align-self: center;
+    color: ${uiColors.gray.base};
+  `;
+}
+
+const ScrollState = {
+  None: 'none',
+  Left: 'left',
+  Right: 'right',
+  Both: 'both',
+};
+
+function getScrollShadowStyle(scrollState: string, variant: Variant): string {
+  const colors = variantColors[variant];
+  const shadowColor =
+    variant === Variant.Light ? 'rgba(93,108,116,0.3)' : 'rgba(0,0,0,0.35)';
+
+  if (scrollState === ScrollState.Both) {
+    return css`
+      box-shadow: inset 6px 0 6px -6px ${shadowColor},
+        inset -6px 0 6px -6px ${shadowColor}, inset 0 6px 6px -6px ${colors[0]},
+        inset 0 -6px 6px -6px ${colors[0]};
+    `;
+  }
+
+  if (scrollState === ScrollState.Left) {
+    return css`
+      box-shadow: inset 6px 0 6px -6px ${shadowColor};
+    `;
+  }
+
+  if (scrollState === ScrollState.Right) {
+    return css`
+      box-shadow: inset -6px 0 6px -6px ${shadowColor};
+    `;
+  }
+
+  return '';
+}
+
+const initialState = {
+  scroll: ScrollState.None,
+  copied: false,
+  isLoaded: false,
+};
+
+function scrollReducer(state: any, action: any) {
```

Comment:
I know these shouldn't be typed as `any` but it's not clear to me what the correct type would be... thoughts?